cls/SourceControl/Git/Util/ProductionConflictResolver.cls

```diff
@@ -0,0 +1,156 @@
+Include (%occInclude, %occErrors, %occKeyword, %occReference, %occSAX)
+
+Class SourceControl.Git.Util.ProductionConflictResolver Extends %RegisteredObject
+{
+
+Property logStream As %Stream.Object [ Private ];
+
+Property productionFile As %String [ Private ];
+
+Property productionClassname As %Dictionary.CacheClassname [ Private ];
+
+Property errorStatus As %Status [ InitialExpression = 1, Private ];
+
+/// API property: whether or not the conflict was resolved
+Property resolved As %Boolean [ InitialExpression = 0 ];
+
+/// API property: error message if resolved is false
+Property errorMessage As %String [ Calculated ];
+
+Method errorMessageGet() As %String
+{
+    If $$$ISERR(..errorStatus) {
+        Do $System.Status.DecomposeStatus(..errorStatus,.components)
+        If $Get(components(1,"code")) = $$$GeneralError {
+            Quit $Get(components(1,"param",1))
+        } Else {
+            Set ex = ##class(%Exception.StatusException).CreateFromStatus(..errorStatus)
+            Do ex.Log()
+            Quit "an internal error occurred and has been logged."
+        }
+    } Else {
+        Quit ""
+    }
+}
+
+ClassMethod FromLog(pOutStream As %Stream.Object) As SourceControl.Git.Util.ProductionConflictResolver
+{
+    Set inst = ..%New()
+    Try {
+        Set inst.logStream = pOutStream
+        Do inst.ConsumeStream()
+        Do inst.Resolve()
+    } Catch e {
+        Set inst.resolved = 0
+        Set inst.errorStatus = e.AsStatus()
+    }
+    Do inst.logStream.Rewind() // Finally
+    Quit inst
+}
+
+Method ConsumeStream() [ Private ]
+{
+    Do ..logStream.Rewind()
+    Do ..logStream.ReadLine()
+    Set productionLine = ..logStream.ReadLine()
+    Set ..productionFile = $Piece(productionLine,"Merge conflict in ",2)
+    If ..productionFile = "" {
+        $$$ThrowStatus($$$ERROR($$$GeneralError,"Message did not reflect merge conflict on a single file."))
+    }
+    If '..logStream.AtEnd {
+        $$$ThrowStatus($$$ERROR($$$GeneralError,"Multiple files had merge conflicts; cannot resolve intelligently."))
+    }
+    Set internalName = ##class(SourceControl.Git.Utils).NameToInternalName(..productionFile)
+    If ($Piece(internalName,".",*) '= "CLS") {
+        $$$ThrowStatus($$$ERROR($$$GeneralError,"File with conflict is not a class."))
+    }
+    Set ..productionClassname = $Piece(internalName,".",1,*-1)
+    If '$$$comClassDefined(..productionClassname) && '$ClassMethod(..productionClassname,"%Extends","Ens.Production") {
```

Comment:
Should this be an or instead of and? Say class with merge conflict is not defined, will try to call %Extends and throw error.

Reply:
Good catch!